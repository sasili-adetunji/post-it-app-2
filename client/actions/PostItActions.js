
import PostItDispatcher from '../dispatcher/PostItDispatcher';
import PostItConstants from '../constants/PostItConstants';

const PostItActions = {
  /**
   * signs in user with & dispatches actions
   * @param {any} user
   */
  login(user) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.LOGIN_USER,
      user
    });
  },

/**
 * signs in user with google & dispatches actions
 * @returns {void}
 */
  googleLogin() {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.GOOGLE_LOGIN
    });
  },

  /**
   * registers in user & dispatches actions
   * @param {any} user
   * @function
   */
  registerUser(user) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.REGISTER_USER,
      user
    });
  },

  /**
   * recieves error message and dispatches actions
   * @param {any} errors
   */
  receiveErrors(errors) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RECEIVE_ERRORS,
      errors
    });
  },

  /**
   * recieves success message and dispatches actions
   * @param {any} message
   */
  receiveSuccess(message) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RECEIVE_SUCCESS,
      message

    });
  },

  /**
   * create group and dispatches actions
   * @param {any} group
   */
  createGroup(group) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.CREATE_GROUP,
      group

    });
  },

  /**
   * add users to group and dispatches an action
   * @param {any} user
   */
  addUserToGroup(user) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.ADDUSER_GROUP,
      user

    });
  },
/**
  * add message and dispatches an action
  * @param {any} message
  */
  addMessage(message) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.ADD_MESSAGE,
      message

    });
  },

  /**
   * resetpassword and dispatches an action
   * @param {any} email
   */
  resetPassword(email) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RESET_PASSWORD,
      email

    });
  },

  /**
   * signout user and dispatches an action
   *
   */
  signOutUser() {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.SIGNOUT_USER

    });
  },

  /**
   * recieve messages and dispatches action
   * @param {any} messages
   */
  receiveMessages(messages) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RECEIVE_MESSAGES,
      messages

    });
  },

  /**
   * recieves user groups and dispatches action
   * @param {any} groups
   */
  receiveUserGroups(groups) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RECEIVE_USER_GROUPS,
      groups

    });
  },

  /**
   * recieves users and dispatches an action
   *
   * @param {any} users
   */
  receiveUsers(users) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RECEIVE_USERS,
      users

    });
  },

  /**
   * recieve authenticated users and dispatches an action
   *
   * @param {any} user
   */
  receiveAuthenticatedUser(user) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.RECEIVE_AUTHENTICATED_USER,
      user

    });
  },

  /**
   * recieves opened group and dispatches an action
   *
   * @param {any} selectedGroup
   */
  groupOpened(selectedGroup) {
    PostItDispatcher.dispatch({
      actionType: PostItConstants.GROUP_OPENED,
      selectedGroup

    });
  }
};

export default PostItActions;
