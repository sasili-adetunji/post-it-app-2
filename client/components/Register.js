import React from 'react';
import mui from 'material-ui';
import { Redirect, browserHistory } from 'react-router-dom';
import { Card, CardActions, CardHeader, CardTitle, CardText } from 'material-ui/Card';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import PostItActions from '../actions/PostItActions';
import PostItStore from '../stores/PostItStore';

/**
 * 
 * 
 * @class Register
 * @extends {React.Component}
 */
class Register extends React.Component {
  /**
   * Creates an instance of Register.
   * @param {object} props
   * @memberOf Register
   */
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      phoneNumber: '',
      isAuthenticated: PostItStore.getIsAuthenticated(),
      errors: PostItStore.getErrors()
    };

    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }
/**
   * Monitors changes in the components and change the state
   * @param {object} e
   * @returns {void}
   * @memberOf Register
*/
  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }
/**
   * Makes an action call to register a user with email and password
   * @param {object} e
   * @returns {void}
   * @memberOf Register
*/
  onClick(e) {
    e.preventDefault();
    const user = {
      email: this.state.email,
      password: this.state.password,
      userName: this.state.userName,
      phoneNumber: this.state.phoneNumber

    };
    PostItActions.registerUser(user);
  }
/**
   * @returns {String} The HTML markup for the Login
   * @memberOf Login
*/
  render() {
    return (
      <div>
        <MuiThemeProvider >
          <Card style={{
            maxWidth: '800px',
            margin: '30px auto',
            padding: '50px',
            textAlign: 'center'
          }}>
            <CardTitle
                style={{ textAlign: 'center' }}
                title="Signup Form"
                subtitle="To continue using PostIt, you need to register below" />
            <TextField
                name="username" onChange={this.onChange}
                value={this.state.userName} hintText="Username Field"
                floatingLabelText="Choose Username" /><br />
            <TextField
                name="email" onChange={this.onChange} value={this.state.email} hintText="Email Field"
                floatingLabelText="Your Email" /><br />
            <TextField
                name="password" onChange={this.onChange}
                value={this.state.password} hintText="Password Field"
                floatingLabelText="Choose Password" type="password" /><br />
            <TextField
                name="phoneNumber" onChange={this.onChange}
                value={this.state.phoneNumber}
                hintText="E.g. 23480" floatingLabelText="Phone Number" /><br />
            <br />
            <span style={{ color: 'red' }} > {PostItStore.getErrors()} </span> <br />
            <p> Already Have an account,<a href="/#/signin"> Login here </a> </p>
            <RaisedButton
                style={{
                  display: 'block',
                }} onClick={this.onClick}
                onTouchTap={this.handleTouchTap}
                label="Sign Up" primary />
          </Card>
        </MuiThemeProvider>
      </div>

    );
  }

}
export default Register;
