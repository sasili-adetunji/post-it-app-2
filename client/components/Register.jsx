import React from 'react';
import PostItActions from '../actions/PostItActions';


/**
 * @description gets user data and register a user
 *
 * @param {object} props
 *
 * @class Register
 *
 * @extends {Component}
 */
class Register extends React.Component {
  /**
  * @description Creates an instance of Register.
  * bind methods and set initial state.
  *
  * @memberof Register
  *
  * @param {object} props
  */
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      email: '',
      password: '',
      phoneNumber: '',
      errors: {}
    };
    this.onClick = this.onClick.bind(this);
    this.onChange = this.onChange.bind(this);
  }


 /**
   * @method onChange
   *
   * @description Monitors changes in the components and change the state
   *
   * @memberof Register
   *
   * @param {SyntheticEvent} event
   *
   * @returns {void}
   */
  onChange(event) {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }


  /**
   * @description Makes an action call to register a user with email and
   * password when all field are filled
   *
   * @param {SyntheticEvent} event
   *
   * @returns {void}
   *
   * @memberof Register
  */
  onClick(event) {
    event.preventDefault();
    const user = {
      email: this.state.email,
      password: this.state.password,
      userName: this.state.userName,
      phoneNumber: this.state.phoneNumber
    };
    if (!user.email) {
      this.setState({ errors: { email: 'Email is required' } });
    } else if (!user.password) {
      this.setState({ errors: { password: 'Password is required' } });
    } else if (!user.phoneNumber) {
      this.setState({ errors: { phoneNumber: 'Phone Number is required' } });
    } else if (!user.userName) {
      this.setState({ errors: { userName: 'userName is required' } });
    } else {
      PostItActions.registerUser(user);

      // clears the state

      this.setState({
        userName: '',
        email: '',
        password: '',
        phoneNumber: '',
        errors: {}
      });
    }
  }


  /**
  * @description Render react component
  *
  * @method render
  *
	* @return { ReactElement } rendered Register page markup
  *
  * @memberof Register
  */
  render() {
    return (
      <div>
        <div className="container">
          <div className="row" />
          <div className="col-md-offset-3 col-md-6">

            <h1>Sign Up</h1>
            <p>To continue using PostIt, you need to sign up below</p>
            <div className="error">
              {this.state.errors.email}
              {this.state.errors.password} {this.state.errors.phoneNumber}
              {this.state.errors.userName}
            </div>
            <form>
              <div className="form-group">
                <label htmlFor="email"> Email address</label>
                <input
                  type="email"
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={this.onChange}
                  value={this.state.email}
                  placeholder="Email"
                />
              </div>
              <div className="form-group">
                <label htmlFor="username"> Username</label>
                <input
                  className="form-control"
                  id="username"
                  name="userName"
                  onChange={this.onChange}
                  value={this.state.userName}
                  placeholder="Username"
                />
              </div>
              <div className="form-group">
                <label htmlFor="phoneNumber">Phone Number</label>
                <input
                  className="form-control"
                  name="phoneNumber"
                  onChange={this.onChange}
                  id="phoneNumber"
                  value={this.state.phoneNumber}
                  placeholder="Phone Number"
                />
              </div>
              <div className="form-group">
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  onChange={this.onChange}
                  value={this.state.password}
                  id="password"
                  placeholder="Password"
                />
              </div>
              <button
                type="submit"
                className="btn btn-primary"
                onClick={this.onClick}
              >Register</button>
            </form>
            <div className="clear" />
            <p> Already have an account? <a href="/#/signin"> Login here
            </a> </p>
          </div>
        </div>
      </div>
    );
  }

}
export default Register;
