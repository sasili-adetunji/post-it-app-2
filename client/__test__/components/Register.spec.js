import React from 'react';
import { shallow, render } from 'enzyme';
import PostItActions from '../../actions/PostItActions'
import Register from '../../components/Register.jsx';


function setup() {
  const props = {
    onClick: () => {},
    onChange: () => {},
    onError: () => {},
    componentDidMount: () => {},
    componentWillUnmount: () => {}
  };
  return shallow(<Register {...props} />);
}


describe('Register', () => {
  it('should match snapshot test', () => {
    const component = shallow(<Register />);
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    const component = shallow(<Register />);
    expect(component).toBeDefined();
  });
  it('Should contain two div', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('Should contain two p', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper.find('p').length).toEqual(1);
  });
  it('Should contain four textfields', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper.find('TextField').length).toEqual(4);
  });
  it('Should contain a Register Button', () => {
    const wrapper = shallow(<Register />);
    expect(wrapper.find('RaisedButton').props().label).toEqual('Sign Up')
    expect(wrapper.find('RaisedButton').length).toEqual(1);
  });
    it('should call function on click of register submit button', () => {
    const component = shallow(<Register />);
    const preventDefault = jest.fn();
    component.find('RaisedButton').simulate('click', { preventDefault });
    expect(preventDefault).toBeCalled();
  });
});
describe('Register  Test', () => {
  it('should take props', () => {
    const wrapper = setup();
    expect(wrapper.props().onChange).toExist;
    expect(wrapper.props().onClick).toExist;
    expect(wrapper.props().onError).toExist;  
    expect(wrapper.props().componentDidMount).toExist;
    expect(wrapper.props().componentWillUnmount).toExist;
  });
});
