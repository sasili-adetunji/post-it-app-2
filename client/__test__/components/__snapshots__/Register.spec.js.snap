// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <MuiThemeProvider>
      <Card
        className="card"
        expandable={false}
        expanded={null}
        initiallyExpanded={false}
      >
        <CardTitle
          subtitle="To continue using PostIt, you need to register below"
          title="Signup Form"
        />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Choose Username"
          fullWidth={false}
          multiLine={false}
          name="userName"
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Your Email"
          fullWidth={false}
          multiLine={false}
          name="email"
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
          value=""
        />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Choose Password"
          fullWidth={false}
          multiLine={false}
          name="password"
          onChange={[Function]}
          rows={1}
          type="password"
          underlineShow={true}
          value=""
        />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Phone Number"
          fullWidth={false}
          multiLine={false}
          name="phoneNumber"
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
        />
        <br />
        <br />
        <p>
           Already Have an account,
          <Link
            replace={false}
            to="/signin"
          >
             Login here 
          </Link>
           
        </p>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Sign Up"
          labelPosition="after"
          onClick={[Function]}
          onTouchTap={undefined}
          primary={true}
          secondary={false}
        />
      </Card>
    </MuiThemeProvider>
  </div>,
  "nodes": Array [
    <div>
      <MuiThemeProvider>
        <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
        >
          <CardTitle
            subtitle="To continue using PostIt, you need to register below"
            title="Signup Form"
          />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Choose Username"
            fullWidth={false}
            multiLine={false}
            name="userName"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Your Email"
            fullWidth={false}
            multiLine={false}
            name="email"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Choose Password"
            fullWidth={false}
            multiLine={false}
            name="password"
            onChange={[Function]}
            rows={1}
            type="password"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Phone Number"
            fullWidth={false}
            multiLine={false}
            name="phoneNumber"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
          />
          <br />
          <br />
          <p>
             Already Have an account,
            <Link
              replace={false}
              to="/signin"
            >
               Login here 
            </Link>
             
          </p>
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Sign Up"
            labelPosition="after"
            onClick={[Function]}
            onTouchTap={undefined}
            primary={true}
            secondary={false}
          />
        </Card>
      </MuiThemeProvider>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Register
      componentDidMount={[Function]}
      componentWillUnmount={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onError={[Function]}
    />,
    "_instance": Register {
      "context": Object {},
      "onChange": [Function],
      "onClick": [Function],
      "onError": [Function],
      "props": Object {
        "componentDidMount": [Function],
        "componentWillUnmount": [Function],
        "onChange": [Function],
        "onClick": [Function],
        "onError": [Function],
      },
      "refs": Object {},
      "state": Object {
        "email": "",
        "errors": "",
        "password": "",
        "phoneNumber": "",
        "userName": "",
      },
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div>
      <MuiThemeProvider>
        <Card
          className="card"
          expandable={false}
          expanded={null}
          initiallyExpanded={false}
        >
          <CardTitle
            subtitle="To continue using PostIt, you need to register below"
            title="Signup Form"
          />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Choose Username"
            fullWidth={false}
            multiLine={false}
            name="userName"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Your Email"
            fullWidth={false}
            multiLine={false}
            name="email"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Choose Password"
            fullWidth={false}
            multiLine={false}
            name="password"
            onChange={[Function]}
            rows={1}
            type="password"
            underlineShow={true}
            value=""
          />
          <br />
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Phone Number"
            fullWidth={false}
            multiLine={false}
            name="phoneNumber"
            onChange={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
          />
          <br />
          <br />
          <p>
             Already Have an account,
            <Link
              replace={false}
              to="/signin"
            >
               Login here 
            </Link>
             
          </p>
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Sign Up"
            labelPosition="after"
            onClick={[Function]}
            onTouchTap={undefined}
            primary={true}
            secondary={false}
          />
        </Card>
      </MuiThemeProvider>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Register
    componentDidMount={[Function]}
    componentWillUnmount={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onError={[Function]}
  />,
}
`;
